astropy/utils/data.py

```diff
@@ -1394,16 +1394,27 @@ def _get_download_cache_locs():
             if not os.path.exists(datadir):
                 raise
     elif not os.path.isdir(datadir):
-        msg = 'Data cache directory {0} is not a directory'
-        raise OSError(msg.format(datadir))
+        raise OSError(f'Data cache directory {datadir} is not a directory')
 
     if os.path.isdir(shelveloc):
-        msg = 'Data cache shelve object location {0} is a directory'
-        raise OSError(msg.format(shelveloc))
+        raise OSError(
+            f'Data cache shelve object location {shelveloc} is a directory')
 
     return datadir, shelveloc
 
 
+def _keep_trying(timeout):
```

Comment:
You did deliver the promise of Python magicky alright! I am learning so much reading your PR. 😄

Reply:
This is the first project where I get to actually assume py36 and I sometimes forget about new (that is, since python 1.5.2) features. So it's fun to take advantage of python's modern tools.